src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -186,13 +189,30 @@ trait SamRecord {
   }
   @inline final def mateUnclippedStart: Option[Int] = {
     require(paired && mateMapped, "Cannot get mate unclipped start position on read without a mapped mate.")
-    get[String]("MC").map(cig => mateStart - Cigar(cig).iterator.takeWhile(_.operator.isClipping).map(_.length).sum)
+    get[String]("MC").map { cig =>
+      val cigar = Cigar(cig)
+      mateStart - cigar.leadingClippedBases
+    }
   }
   @inline final def mateUnclippedEnd: Option[Int] = {
     require(paired && mateMapped, "Cannot get mate unclipped end position on read without a mapped mate.")
     get[String]("MC").map { cig =>
       val cigar = Cigar(cig)
-      mateStart + cigar.lengthOnTarget - 1 + cigar.reverseIterator.takeWhile(_.operator.isClipping).map(_.length).sum
+      mateStart + cigar.lengthOnTarget - 1 + cigar.trailingClippedBases
+    }
+  }
+  @inline final def mateUnSoftClippedStart: Option[Int] = {
+    require(paired && mateMapped, "Cannot get mate un-soft-clipped start position on read without a mapped mate.")
+    get[String]("MC").map { cig =>
+      val cigar = Cigar(cig)
+      mateStart - cigar.leadingSoftClippedBases
+    }
+  }
+  @inline final def mateUnSoftClippedEnd: Option[Int] = {
+    require(paired && mateMapped, "Cannot get mate un-soft-clipped end position on read without a mapped mate.")
+    get[String]("MC").map { cig =>
+      val cigar = Cigar(cig)
+      mateStart + cigar.lengthOnTarget - 1 + cigar.trailingSoftClippedBases
     }
   }
```

Comment:
I'm starting to think we should cache the mate cigar like we do with the record's cigar. It would require similar trickery to detect when a new mate cigar is updated upon the record since we would have to intercept any call to `setAttribute`, check if the attribute is being set with the `MC` tag, and then reset the in-memory mate cigar to the new value. If I implement this suggestion, it will be in a separate PR.

Reply:
https://github.com/fulcrumgenomics/fgbio/issues/1115